### ISSUE TITLE
[Question] Why threads needs to end in a divert, insted of "go on"

### ISSUE THREAD

**Issue by rubereaglenest** [inkle/ink#318]
Hi, so... I'm using threads, and yes, they are so complicated as they sound. But I want to ask, Why they need to return with a divert insted of a ->-> ?

**Comment by joningold** [comment]
Because they aren't tunnels at all; they can't and don't "return". You
should think about them not as a function call, but as though the normal
story flow split into two alternative future timelines; and when a choice
is made in one the other vanishes, but until a choice is made both are
equally valid and neither is the "real."

Jon

**Comment by JoshuaGrams** [comment]
I feel like the documentation around threads is confusing for no particular reason. Like maybe it's bad programmer-documentation? Describing the implementation rather than describing its behavior in user-facing terms.

I've been thinking of `<- name` as simply merging the first "beat" of the named knot/stitch into the current one (where a "beat" is the core unit of Ink gameplay: a piece of story text followed by a set of choices -- do you guys have a name for this?).  That is, replace `<- name` with its story text, add its options to the current set of choices.

That seems like a much simpler model, and I haven't found any loopholes in that metaphor yet...am I missing something?

**Comment by joningold** [comment]
No, that metaphor works. I think the one thing that ought to be stressed
though is that a thread both merges in content -- and diverges. If you pick
an option in that thread, there's no way back to the "head".

cheers
jon

**Comment by mwchase** [comment]
I don't know if this helps any in explaining things, but I've used Ink some to represent the actions a user can take in an application, and threads are a really natural fit for stuff like, say, a sidebar in a website that's always visible.
